src/Messages/OWSDisappearingMessagesJob.m

```diff
@@ -84,16 +88,20 @@ - (void)run
     uint64_t now = [NSDate ows_millisecondTimeStamp];
 
     __block uint expirationCount = 0;
-    [self.disappearingMessagesFinder enumerateExpiredMessagesWithBlock:^(TSMessage *message) {
-        // sanity check
-        if (message.expiresAt > now) {
-            DDLogError(@"%@ Refusing to remove message which doesn't expire until: %lld", self.tag, message.expiresAt);
-            return;
+    [self.storageManager.dbConnection readWriteWithBlock:^(YapDatabaseReadWriteTransaction *_Nonnull transaction) {
```

Comment:
I'm starting to feel that having a global dbConnection on storageManager might be an anti-pattern.

Previously the finder had it's own dbConnection (via `newDatabaseConnection`) which means it's operations had their own serial queue. Since that's been removed, I think this job should have it's own dbConnection if we don't want it to interfere with(/be interfered with) other things that share the storageManager.dbConnection's serial queue.